examples/gallery/mvdr/mvdr_tutorial.py

```diff
@@ -0,0 +1,291 @@
+#    square root of it. This is slightly different from the definition of IRM.
+# 
+
+
+def get_irms(spec_clean, spec_noise, spec_mix):
+    mag_mix = spec_mix.abs() ** 2
+    mag_clean = spec_clean.abs() ** 2
+    mag_noise = spec_noise.abs() ** 2
+    irm_speech = mag_clean / (mag_clean + mag_noise)
+    irm_noise = mag_noise / (mag_clean + mag_noise)
+
+    return irm_speech, irm_noise
+
+######################################################################
+# .. note::
+#    We use reverberant clean speech as the target here,
+#    you can also set it to dry clean speech.
+
+irm_speech, irm_noise = get_irms(spec_reverb_clean, spec_noise, spec_mix)
+
+######################################################################
+# Apply MVDR
+# ==========
+# 
+
+######################################################################
+# Apply MVDR beamforming by using multi-channel masks
+# ---------------------------------------------------
+# 
+
+results_multi = {}
+for solution in ['ref_channel', 'stv_evd', 'stv_power']:
+    mvdr = torchaudio.transforms.MVDR(ref_channel=0, solution=solution, multi_mask=True)
+    stft_est = mvdr(spec_mix, irm_speech, irm_noise)
+    est = istft(stft_est, length=mix.shape[-1])
+    results_multi[solution] = est
+
+######################################################################
+# Apply MVDR beamforming by using single-channel masks
+# ----------------------------------------------------
+# 
+# We use the 1st channel as an example.
+# The channel selection may depend on the design of the microphone array
+
+results_single = {}
+for solution in ['ref_channel', 'stv_evd', 'stv_power']:
+    mvdr = torchaudio.transforms.MVDR(ref_channel=0, solution=solution, multi_mask=False)
+    stft_est = mvdr(spec_mix, irm_speech[0], irm_noise[0])
+    est = istft(stft_est, length=mix.shape[-1])
+    results_single[solution] = est
+
+######################################################################
+# Compute Si-SDR scores
+# ---------------------
+# 
+
+def si_sdr(estimate, reference, epsilon=1e-8):
+    estimate = estimate - estimate.mean()
+    reference = reference - reference.mean()
+    reference_pow = reference.pow(2).mean(axis=1, keepdim=True)
+    mix_pow = (estimate * reference).mean(axis=1, keepdim=True)
+    scale = mix_pow / (reference_pow + epsilon)
+
+    reference = scale * reference
+    error = estimate - reference
+
+    reference_pow = reference.pow(2)
+    error_pow = error.pow(2)
+
+    reference_pow = reference_pow.mean(axis=1)
+    error_pow = error_pow.mean(axis=1)
+
+    sisdr = 10 * torch.log10(reference_pow) - 10 * torch.log10(error_pow)
+    return sisdr.item()
+
+######################################################################
+# Results
+# =======
+# 
+
+######################################################################
+# Single-channel mask results
+# ---------------------------
+# 
+
+for solution in results_single:
```

Comment:
I feel like this is better listed with the audio sample (and maybe with spectrogram). What do you think?

Reply:
Yeah, that's a good idea.